composer/utils/batch_helpers.py

```diff
@@ -0,0 +1,211 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Callable, Optional, Sequence, Union
+
+__all__ = ['batch_get', 'batch_set']
+
+
+def batch_get(batch: Any, key: Optional[Any] = None, get_fn: Optional[Callable] = None) -> Any:
```

Comment:
I like how this API style is explicit, but just want to confirm how an algorithm would call this?

```python

class MyAlg:
  def __init__(self, key: Any = None, get_fn: Optional[Callable[[], Any]] = None, set_fn: Optional[Callable[[str, Any], None]] = None):
    self.key, self.get_fn, self.set_fn = key, get_fn, set_fn

  def apply(self, state, event):
    batch_element = state.get_item_in_batch(self.key, self.get_fn)
    batch_element = apply_alg_magic(batch_element)
    state.set_item_in_batch(key=self.key, value=batch_element, set_fn=self.set_fn)
```

Reply:
Yeah that looks right

Reply:
I guess that means the set_fn would have to take a value as input. Let me change that!

Reply:
Good call!

Reply:
fixed here: 
55ace59